### PR TITLE
Hw5 t01

### PR DESCRIPTION
**Pull request by mirmozavr**
Homework, Student, Teacher

### CODE REVIEW

File: hw5/hw5_t01.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+1. Homework принимает на вход 2 атрибута: текст задания и количество дней
+на это задание
+Атрибуты:
+    text - текст задания
+    deadline - хранит объект datetime.timedelta с количеством
+    дней на выполнение
+    created - c точной датой и временем создания
+Методы:
+    is_active - проверяет не истело ли время на выполнение задания,
```
Comment (Cjkjvfnby) [inline hw5/hw5_t01.py]: `не истело ли время ` -> `не истлело ли время`

File: hw5/hw5_t01.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+давать логичные подходящие имена.
+"""
+import datetime
+from typing import Any
+
+
+class Homework:
+    """Homework class."""
+
+    def __init__(self, text: str, deadline: int):
```
Comment (Cjkjvfnby) [inline hw5/hw5_t01.py]: Why deadline is int? 

This name does not contain information on how to measure it. Is it second, or hours, or maybe milliseconds?

File: hw5/hw5_t01.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+
+class Student:
+    """Student class."""
+
+    def __init__(self, last_name: str, first_name: str):
+        self.last_name = last_name
+        self.first_name = first_name
+
+    def do_homework(self, homework: Homework) -> Any:
```
Comment (Cjkjvfnby) [inline hw5/hw5_t01.py]: Plesse check `typing.Union` and `typing.Optional`

File: hw5/tests/test_hw5_t01.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import datetime
+
+from hw5.hw5_t01 import Homework, Student, Teacher
+
+
+def test_attributes_of_teacher_class():
+    teacher = Teacher("Smith", "Adam")
+    assert teacher.last_name, teacher.first_name == ("Smith", "Adam")
```
Comment (Cjkjvfnby) [inline hw5/tests/test_hw5_t01.py]: You don't need parenthesis.
I'd prefer putting checks to separate lines.

File: hw5/tests/test_hw5_t01.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    student = Student("Doe", "John")
+    assert student.last_name, student.first_name == ("Doe", "John")
+
+
+def test_attributes_of_homework_class():
+    homework = Homework("Tasks", 3)
+    assert homework.text, homework.deadline == ("Tasks", 3)
+
+
+def test_negative_is_active_method_of_homework_class():
```
Comment (Cjkjvfnby) [inline hw5/tests/test_hw5_t01.py]: negative is active method of homeworkclass, isn't it?

`test_expaired_homework_is_not_active`

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
Can you make test names better?

File: hw5/hw5_t01.py (hunk 2 above)
Comment (mirmozavr) [reply hw5/hw5_t01.py]: We are accepting an integer number of days and then we store it as `datetime.timedelta`.

File: hw5/tests/test_hw5_t01.py (hunk 5 above)
Comment (mirmozavr) [reply hw5/tests/test_hw5_t01.py]: No pun intended XD

File: hw5/hw5_t01.py (hunk 2 above)
Comment (Cjkjvfnby) [reply hw5/hw5_t01.py]: > This name does not contain information on how to measure it. 

Can we solve it?

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**